gui/WorldView.cxx

```diff
@@ -473,123 +469,122 @@ void WorldView::setBackgroundColor(QColor const& color)
 }
 
 //-----------------------------------------------------------------------------
-void WorldView::setActiveDepthMap(vtkMaptkCamera* camera, QString vtiPath)
+void WorldView::setActiveDepthMap(
+  vtkMaptkCamera* camera, QString const& depthMapPath)
 {
   QTE_D();
 
-  if(d->UI.actionDepthmapDisplay->isChecked() && !d->depthmapAlreadLoaded)
+  if (d->UI.actionShowDepthMap->isChecked() && !d->depthMapLoaded)
   {
-    std::string filename = vtiPath.toStdString();
+    vtkNew<vtkXMLImageDataReader> reader;
 
-    vtkNew<vtkXMLImageDataReader> readerIm;
+    reader->SetFileName(qPrintable(depthMapPath));
+    reader->Update();
 
-    readerIm->SetFileName(filename.c_str());
-    readerIm->Update();
+    auto const pointCount = reader->GetOutput()->GetNumberOfPoints();
+    if (!pointCount)
+    {
+      qWarning() << "Failed to read depth map from" << depthMapPath;
+      return;
+    }
 
-    int width = readerIm->GetOutput()->GetDimensions()[0];
-    int height = readerIm->GetOutput()->GetDimensions()[1];
+    auto const width = reader->GetOutput()->GetDimensions()[0];
+    auto const height = reader->GetOutput()->GetDimensions()[1];
+    qDebug() << "point counts:" << width * height << pointCount;
```

Comment:
Oops, that snuck in by mistake; sorry :smile:. @mleotta, if this is the only issue, it may be quicker to merge anyway and then just nuke it right from master...